src/components/Trait/WardrobeTraitSelector.tsx

```diff
@@ -49,11 +50,23 @@ const WardrobeTraitSelector: React.FC<Props> = ({ categories }) => {
 		value: string,
 	) => {
 		const hadMatch = metadataTraits.find(t => t.name === value)
-		const updatedMetadata = metadataTraits.filter(
-			t => t.categories?.join(' - ') !== categoryString,
-		)
+		const mustHaveExactlyOne = oneAndOnlyOne.includes(categoryString)
+		let updatedMetadata = [...metadataTraits]
+
+		if (hadMatch && mustHaveExactlyOne) {
+			// Can't remove this trait
+			return
```

Comment:
This is a quiet return. 
@misskatles When a trait is selected and you are required to have exactly 1 of that kind, how do we display that info to the user?

Reply:
In a game - when you can't take off a trait what does it do? 

In Palworld it just doesn't change - There is no 'none' option and clicking the trait twice doesn't do anything.  It only changes when they click on a different trait. 

If you are talking about clothes - could make a warning pop up regarding it being removed or just have it default to something else. They are just a stick figure underneath still  🤷‍♀️

Reply:
This actually raises another question. Does this mean that they can't be mixed and matched (you can only have one item from each of the below four).

    Clothing - Tops && Clothing - Bottoms OR
    Clothing - Set OR
    Clothing - Dresses OR
    Clothing - Jumpsuits

Ie. you can't have a Top with a Dress.

Reply:
@foopod - No you could still mix and match.  By virtue of you having a dress you satisfy the validation.

So the only one where you would have to have both to be valid is the Tops & Bottoms>
Examples of clothing valid conditions:

Top & Bottom = Top & Bottom valid condition met
Top & Bottom & Outerwear = Top & Bottom valid condition met
Top & Dress = Dress valid condition met
Dress & Outerwear = Dress valid condition met
Dress & Bottoms = Dress valid condition met
Jumpsuit & Accessories = Jumpsuit valid condition met
Jumpsuit & Top & Bottoms = Jumpsuit + Top & Bottoms valid conditions met
Jumpsuit & Top = Jumpsuit valid condition met
Set & Top = Set valid condition met
etc

Reply:
@misskatles Could you also mix and match within a category? For example two tops eg. Safety vest over another top, or two bottoms eg. Leggings under shorts.

Reply:
Yup! So a short sleeve can layer on top of a long sleeve. Some might not be as visible like the tops - for example leggings under jeans, the leggings wouldn't be visible on the Peep but it could still be selected.

Reply:
Could there be like a trait counter?  I feel like that could be fun.  See how many pieces of clothing your peep could wear lol

Reply:
> Yup! So a short sleeve can layer on top of a long sleeve. Some might not be as visible like the tops - for example leggings under jeans, the leggings wouldn't be visible on the Peep but it could still be selected. 

Awesome. This actually makes things simple on the coding side then.

The rules is essentially "must have some clothing", no limits for now on how much or how you mix and match.